src/utils/permalinks/Permalinks.ts

```diff
@@ -96,7 +96,7 @@ export class RoomPermalinkCreator {
     // Some of the tests done by this class are relatively expensive, so normally
     // throttled to not happen on every update. Pass false as the shouldThrottle
     // param to disable this behaviour, eg. for tests.
-    public constructor(private room: Room, roomId: string | null = null, shouldThrottle = true) {
+    public constructor(private room: Room | null, roomId: string | null = null, shouldThrottle = true) {
```

Comment:
although this looks correct, it looks to have exposed some typescript check failures. Probably some more work to do here to fix things up.

Reply:
Fix attempted in 5a110fdb81